From dfbd7b3402e46f4f3140cb56c0f1f499aaf0f448 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Sun, 15 Feb 2015 10:25:58 +0800
Subject: [PATCH 0132/1221] MLK-10278-3 ARM: dts: imx7d: add epdc setting

Add EPDC setting, including iomux, display pmic setting, clock setting,
and GPR setting.

Signed-off-by: Robby Cai <r63905@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts |   92 ++++++++++++++++++--------------
 arch/arm/boot/dts/imx7d.dtsi           |    3 +
 2 files changed, 54 insertions(+), 41 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 6209482..2a97398 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -126,6 +126,15 @@
 	};
 };
 
+&epdc {
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_epdc_0>;
+        V3P3-supply = <&V3P3_reg>;
+        VCOM-supply = <&VCOM_reg>;
+        DISPLAY-supply = <&DISPLAY_reg>;
+        status = "okay";
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
@@ -167,12 +176,13 @@
 	hog {
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
-				MX7D_PAD_I2C4_SCL__GPIO4_IO14	0x80000000
-				MX7D_PAD_EPDC_PWR_STAT__GPIO2_IO31 0x80000000
-				MX7D_PAD_ECSPI2_SCLK__GPIO4_IO20  0x80000000
-				MX7D_PAD_ECSPI2_MOSI__GPIO4_IO21  0x80000000
-				MX7D_PAD_ECSPI2_MISO__GPIO4_IO22  0x80000000
-				MX7D_PAD_ECSPI2_SS0__GPIO4_IO23	  0x80000000
+				MX7D_PAD_EPDC_PWR_STAT__GPIO2_IO31 0x80000000 /* pwrgood */
+				MX7D_PAD_I2C4_SCL__GPIO4_IO14     0x80000000  /* vcom_ctrl */
+				MX7D_PAD_ECSPI2_SS0__GPIO4_IO23	  0x80000000  /* wakeup */
+				MX7D_PAD_ECSPI2_SCLK__GPIO4_IO20  0x80000000  /* v3p3 */
+				MX7D_PAD_ECSPI1_MISO__GPIO4_IO18  0x80000000  /* pwr int */
+				MX7D_PAD_ECSPI2_MOSI__GPIO4_IO21  0x80000000  /* pwrctrl1 */
+				MX7D_PAD_ECSPI2_MISO__GPIO4_IO22  0x80000000  /* pwrctrl2 */
 				MX7D_PAD_ECSPI1_MOSI__GPIO4_IO17  0x80000000
 				MX7D_PAD_SD1_RESET_B__GPIO5_IO2   0x59
 				MX7D_PAD_SD1_CD_B__GPIO5_IO0      0x59
@@ -335,36 +345,36 @@
 	epdc {
                 pinctrl_epdc_0: epdcgrp-0 {
                         fsl,pins = <
-				MX7D_PAD_EPDC_DATA00__EPDC_DATA0 0x80000000
-				MX7D_PAD_EPDC_DATA01__EPDC_DATA1 0x80000000
-				MX7D_PAD_EPDC_DATA02__EPDC_DATA2 0x80000000
-				MX7D_PAD_EPDC_DATA03__EPDC_DATA3 0x80000000
-				MX7D_PAD_EPDC_DATA04__EPDC_DATA4 0x80000000
-				MX7D_PAD_EPDC_DATA05__EPDC_DATA5 0x80000000
-				MX7D_PAD_EPDC_DATA06__EPDC_DATA6 0x80000000
-				MX7D_PAD_EPDC_DATA07__EPDC_DATA7 0x80000000
-				MX7D_PAD_EPDC_DATA08__EPDC_DATA8 0x80000000
-				MX7D_PAD_EPDC_DATA09__EPDC_DATA9 0x80000000
-				MX7D_PAD_EPDC_DATA10__EPDC_DATA10 0x80000000
-				MX7D_PAD_EPDC_DATA11__EPDC_DATA11 0x80000000
-				MX7D_PAD_EPDC_DATA12__EPDC_DATA12 0x80000000
-				MX7D_PAD_EPDC_DATA13__EPDC_DATA13 0x80000000
-				MX7D_PAD_EPDC_DATA14__EPDC_DATA14 0x80000000
-				MX7D_PAD_EPDC_DATA15__EPDC_DATA15 0x80000000
-				MX7D_PAD_EPDC_SDCLK__EPDC_SDCLK 0x80000000
-				MX7D_PAD_EPDC_SDLE__EPDC_SDLE 0x80000000
-				MX7D_PAD_EPDC_SDOE__EPDC_SDOE 0x80000000
-				MX7D_PAD_EPDC_SDSHR__EPDC_SDSHR 0x80000000
-				MX7D_PAD_EPDC_SDCE0__EPDC_SDCE0 0x80000000
-				MX7D_PAD_EPDC_SDCE1__EPDC_SDCE1 0x80000000
-				MX7D_PAD_EPDC_SDCE2__EPDC_SDCE2 0x80000000
-				MX7D_PAD_EPDC_SDCE3__EPDC_SDCE3 0x80000000
-				MX7D_PAD_EPDC_GDCLK__EPDC_GDCLK 0x80000000
-				MX7D_PAD_EPDC_GDOE__EPDC_GDOE 0x80000000
-				MX7D_PAD_EPDC_GDRL__EPDC_GDRL 0x80000000
-				MX7D_PAD_EPDC_GDSP__EPDC_GDSP 0x80000000
-				MX7D_PAD_EPDC_BDR0__EPDC_BDR0 0x80000000
-				MX7D_PAD_EPDC_BDR1__EPDC_BDR1 0x80000000
+				MX7D_PAD_EPDC_DATA00__EPDC_DATA0  0x2
+				MX7D_PAD_EPDC_DATA01__EPDC_DATA1  0x2
+				MX7D_PAD_EPDC_DATA02__EPDC_DATA2  0x2
+				MX7D_PAD_EPDC_DATA03__EPDC_DATA3  0x2
+				MX7D_PAD_EPDC_DATA04__EPDC_DATA4  0x2
+				MX7D_PAD_EPDC_DATA05__EPDC_DATA5  0x2
+				MX7D_PAD_EPDC_DATA06__EPDC_DATA6  0x2
+				MX7D_PAD_EPDC_DATA07__EPDC_DATA7  0x2
+				MX7D_PAD_EPDC_DATA08__EPDC_DATA8  0x2
+				MX7D_PAD_EPDC_DATA09__EPDC_DATA9  0x2
+				MX7D_PAD_EPDC_DATA10__EPDC_DATA10 0x2
+				MX7D_PAD_EPDC_DATA11__EPDC_DATA11 0x2
+				MX7D_PAD_EPDC_DATA12__EPDC_DATA12 0x2
+				MX7D_PAD_EPDC_DATA13__EPDC_DATA13 0x2
+				MX7D_PAD_EPDC_DATA14__EPDC_DATA14 0x2
+				MX7D_PAD_EPDC_DATA15__EPDC_DATA15 0x2
+				MX7D_PAD_EPDC_SDCLK__EPDC_SDCLK   0x2
+				MX7D_PAD_EPDC_SDLE__EPDC_SDLE     0x2
+				MX7D_PAD_EPDC_SDOE__EPDC_SDOE     0x2
+				MX7D_PAD_EPDC_SDSHR__EPDC_SDSHR   0x2
+				MX7D_PAD_EPDC_SDCE0__EPDC_SDCE0   0x2
+				MX7D_PAD_EPDC_SDCE1__EPDC_SDCE1   0x2
+				MX7D_PAD_EPDC_SDCE2__EPDC_SDCE2   0x2
+				MX7D_PAD_EPDC_SDCE3__EPDC_SDCE3   0x2
+				MX7D_PAD_EPDC_GDCLK__EPDC_GDCLK   0x2
+				MX7D_PAD_EPDC_GDOE__EPDC_GDOE     0x2
+				MX7D_PAD_EPDC_GDRL__EPDC_GDRL     0x2
+				MX7D_PAD_EPDC_GDSP__EPDC_GDSP     0x2
+				MX7D_PAD_EPDC_BDR0__EPDC_BDR0     0x2
+				MX7D_PAD_EPDC_BDR1__EPDC_BDR1     0x2
                         >;
                 };
 	};
@@ -709,11 +719,11 @@
 		vpos_pwrdn = <2>;
 		gvee_pwrdn = <8>;
 		vneg_pwrdn = <10>;
-		gpio_pmic_pwrgood = <&gpio2 13 0>;
-		gpio_pmic_vcom_ctrl = <&gpio2 3 0>;
-		gpio_pmic_wakeup = <&gpio2 14 0>;
-		gpio_pmic_v3p3 = <&gpio2 7 0>;
-		gpio_pmic_intr = <&gpio2 12 0>;
+		gpio_pmic_pwrgood = <&gpio2 31 0>;
+		gpio_pmic_vcom_ctrl = <&gpio4 14 0>;
+		gpio_pmic_wakeup = <&gpio4 23 0>;
+		gpio_pmic_v3p3 = <&gpio4 20 0>;
+		gpio_pmic_intr = <&gpio4 18 0>;
 
 		regulators {
 			DISPLAY_reg: DISPLAY {
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 1ecae35..8614fcb 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -691,6 +691,9 @@
 				compatible = "fsl,imx7d-epdc";
 				interrupts = <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>;
 			        reg = <0x306f0000 0x10000>;
+				clocks = <&clks IMX7D_CLK_DUMMY>, <&clks IMX7D_EPDC_PIXEL_ROOT_CLK>;
+				clock-names = "epdc_axi", "epdc_pix";
+				epdc-ram = <&gpr 0x4 30>;
 				status = "disabled";
 			};
 
-- 
1.7.5.4

